### PR TITLE
Refactored scaffold build

### PR DESCRIPTION
**Pull request by U-DON**
## Description
- Updated the watch script to use the scaffold option by default.
- Refactored the scaffolding logic into its own function.
- Allowed the scaffold to work without a local `content-build` by downloading the necessary files directly.
  - This enables building apps without having to check out all of `content-build` (possibly for CI).
  - If there is a local `content-build`, the scaffold will use those files instead.

### Downloading external assets (no local `content-build`)
```
$ NODE_OPTIONS=--max-old-space-size=4096 webpack --config config/webpack.config.js -n --env.scaffold
Created directory for scaffold assets: tmp/scaffold
Downloading asset from https://raw.githubusercontent.com/department-of-veterans-affairs/content-build/master/src/applications/registry.json.
Downloading asset from https://raw.githubusercontent.com/department-of-veterans-affairs/content-build/master/src/site/assets/js/incompatible-browser.js.
Downloading asset from https://raw.githubusercontent.com/department-of-veterans-affairs/content-build/master/src/site/assets/js/record-event.js.
Downloading asset from https://raw.githubusercontent.com/department-of-veterans-affairs/content-build/master/src/site/assets/js/static-page-widgets.js.
Saved asset to tmp/scaffold/static-page-widgets.js.
Saved asset to tmp/scaffold/registry.json.
Saved asset to tmp/scaffold/record-event.js.
Saved asset to tmp/scaffold/incompatible-browser.js.
```

### Pulling local assets (with local `content-build`)
```
$ NODE_OPTIONS=--max-old-space-size=4096 webpack --config config/webpack.config.js -n --env.scaffold
Created directory for scaffold assets: tmp/scaffold
Linked to local asset found at /Users/eugenedoan/Code/content-build/src/applications/registry.json.
Linked to local asset found at /Users/eugenedoan/Code/content-build/src/site/assets/js/incompatible-browser.js.
Linked to local asset found at /Users/eugenedoan/Code/content-build/src/site/assets/js/record-event.js.
Linked to local asset found at /Users/eugenedoan/Code/content-build/src/site/assets/js/static-page-widgets.js.
```

## Testing done
Tested building and watching with and without `content-build` as a sibling directory.

## Acceptance criteria
- [ ] Build should work without local `content-build`.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: config/webpack.config.js [hunk 1]
```diff
@@ -84,7 +87,170 @@ function getEntryPoints(entry) {
   return getWebpackEntryPoints(manifestsToBuild);
 }
 
-module.exports = (env = {}) => {
+async function generateHtmlFiles(buildPath) {
+  const TMP_SCAFFOLD_PATH = 'tmp/scaffold';
+  fs.removeSync(TMP_SCAFFOLD_PATH);
+  fs.ensureDirSync(TMP_SCAFFOLD_PATH);
```
Comment (theodur) [inline config/webpack.config.js:93]: Do we still need to make the `tmp` directory if the `content-build` directory exists locally? I noticed that the `tmp` directory with all the downloaded files are generated, despite having `content-build` cloned locally.

File: jenkins/common.groovy [hunk 2]
```diff
@@ -74,15 +74,6 @@ def slackIntegrationNotify() {
 
 def setup() {
   stage("Setup") {
-
-    dir("vagov-content") {
```
Comment (theodur) [inline jenkins/common.groovy:78]: This is unrelated to this PR, but this reminded me that we can probably remove `vagov-content` from the `install-repos.sh` script.

File: config/webpack.config.js (hunk 1 above)
Comment (U-DON) [reply config/webpack.config.js:93]: Probably not, but for the sake of keeping the code simple and the flow consistent, I figured it should be fine to read from the same place.

File: jenkins/common.groovy (hunk 2 above)
Comment (timwright12) [reply jenkins/common.groovy:78]: I think CMS team uses that, I would check with them first

**Review by timwright12 — APPROVED** [review]
Pipeline is clean, file-manifest matches prod, and the site appears to work locally. I'm fine with this update. We do need to address the size of this file though. There's a ticket open for it https://github.com/department-of-veterans-affairs/va.gov-team/issues/2796

**Review by theodur — APPROVED** [review]
🚀

**Pull request merged**